doc/reference/plutus/plutus-spending-script-example.md

```diff
@@ -0,0 +1,119 @@
+
+The transaction execution units are an upper bound or budget of what will be spent to execute the Plutus script. If you don't specify a high enough value to cover script execution, your transaction will still be successful (provided it is a valid tx) but you will lose your collateral. The collateral is the transaction input(s) you specify to be consumed if your script fails to execute. There is a protocol parameter `collateralPercent` that determines what percentage of your inputs you must supply as collateral.
+
+*Note that in order to use a tx input as collateral, it **cannot** reside at a script address; it must reside at a ‘normal’ payment address and it cannot contain any multi-assets.*
+
+### An example of using a Plutus spending script
+
+Below is an example that shows how to use a Plutus spending script. This is a step-by-step
+process involving:
+
++ the creation of the `AlwaysSucceeds` Plutus txin script
++ sending ADA to the Plutus script address
++ spending ADA at the Plutus script address
+
+In this example we will use the [AlwaysSucceeds](../../../plutus-example/plutus-example/src/Cardano/PlutusExample/AlwaysSucceeds.hs) Plutus spending script. In order to execute a Plutus spending script, we require the following:
+
+- Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx output that sits at the Plutus script address. It must have a datum hash, otherwise, it is unspendable.
+- The Plutus script serialized in the text envelope format. `cardano-cli` expects Plutus scripts to be serialized in the text envelope format.
+
+#### Creating the `AlwaysSucceeds` Plutus spending script
+
+The plutus-example executable will automagically generate several Plutus scripts in the CLI-compatiable text envelope format.
+
+Run the following commands:
+
+```bash
+cd plutus-example
+
+cabal run exe:plutus-example
+```
+
+This will output `always-succeeds-txin.plutus` in the `generated-plutus-scripts` dir.
+
+#### Setting up a local Alonzo node cluster
+
+There is a convenient script that will set up an Alonzo cluster immediately on your local machine.
+
+Run the following command:
+
+```bash
+cabal install cardano-cli
+cabal install cardano-node
+./scripts/byron-to-alonzo/mkfiles.sh alonzo
+```
+
+Follow the instructions displayed in the terminal to start your Alonzo cluster.
+
+#### Sending ADA to the script address
+
+In order to require a Plutus script to validate the spending of a tx ouput, we must put the tx output at the script address of the said Plutus script. However, before we do that, we must create a datum hash:
+
+```bash
+> cardano-cli transaction hash-script-data --script-data-value 42
+> 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b
```

Comment:
I'm wondering if we're already using `42` for testnet magic, if we should use another number for datum hash?

Reply:
We can but I don't think its an issue.